src/imageio/imageio_dng.h

```diff
@@ -38,22 +38,28 @@
 #define RATIONAL 5
 #define SRATIONAL 10
 
-static inline void dt_imageio_dng_write_buf(uint8_t *buf, int adr, int val)
+#define HEADBUFFSIZE 1024
+
+static inline void dt_imageio_dng_write_buf(uint8_t *buf, int d, int val)
 {
-  buf[adr + 3] = val & 0xff;
-  buf[adr + 2] = (val >> 8) & 0xff;
-  buf[adr + 1] = (val >> 16) & 0xff;
-  buf[adr] = val >> 24;
+  if(d + 4 >= HEADBUFFSIZE) return;
+  buf[d + 3] = val & 0xff;
+  buf[d + 2] = (val >> 8) & 0xff;
+  buf[d + 1] = (val >> 16) & 0xff;
+  buf[d] = val >> 24;
 }
 
-static inline uint8_t *dt_imageio_dng_make_tag(
+static inline int dt_imageio_dng_make_tag(
```

Comment:
as we are in code maintenance bug, `uint32_t` seems more appropriate here maybe.

Reply:
Yes, i agree. There is also the int data later on in the code that could/should be unsigned too...